### PR TITLE
Add Lecture 7

### PR DESCRIPTION
**Pull request by varkor**
This provides at least the skeleton of Lecture 7's material. It was diagram-heavy, so some reformatting might be desirable, but this gives a good starting point.

### CODE REVIEW

File: acs-category-theory-notes.tex [hunk 1]
```diff
@@ -2497,7 +2506,267 @@ \subsection{Morphisms of monoids, preorders, and categories}
...
 
-\chapter{Natural transformations}
-\lecturedetails{26 October 2017}{M Fiore, \ldots}
+\chapter{Functors continued}
+\lecturedetails{26 October 2017}{M Fiore, N Alcock, Z Yuan}
+
+\section{More about functors}
+
+\begin{definition}
+A functor \(F: \mathcal{A} \rightarrow \mathcal{B}\) specifies a function \(F: \Ob(\mathcal{A}) \rightarrow \Ob(\mathcal{B})\) such that \(\forall X, Y \in \Ob(\mathcal{A})\), there exists \(F_{X, Y}: \mathcal{A}(X, Y) \rightarrow \mathcal{B}(FX, FY)\) such that identities are preserved.
```
Comment (dc-mak) [inline acs-category-theory-notes.tex:2515]: I think " mapping " instead of " function "? Open to discussion.

File: acs-category-theory-notes.tex [hunk 2]
```diff
@@ -2497,7 +2506,267 @@ \subsection{Morphisms of monoids, preorders, and categories}
...
+Show that the free commutative monoid (\ie~finite multisets) construction is a functor. \label{ex:fcmonoidfunctor}
+\end{exercise}
+
+\begin{exercise}
+Show that the free semilattice (commutative idempotent monoids --- \ie finite powersets) construction is a functor. \label{ex:fsemilatticefunctor}
+\end{exercise}
+
+\subsection{More examples: product and coproduct constructions}
+
+The (locally small) category \(\catcat\) has as objects small categories, \(\mathbb{C}\), and as morphisms functors \(\mathbb{C} \morpharr{F} \mathbb{D}\). Identities are \(\Id{\mathbb{C}}\) and composition \(\mathbb{A} \morpharr{F} \mathbb{B} \morpharr{G} \mathbb{C}\) such that \((G \comp F)(A) \eqdef G(FA)\).
```
Comment (dc-mak) [inline acs-category-theory-notes.tex:2705]: Can you make this a definition?

**Review by dc-mak — APPROVED** [review]
Thanks for the work! I'll merge for now and the changes can be made in a different pull request.

File: acs-category-theory-notes.tex (hunk 1 above)
Comment (technicalguy) [reply acs-category-theory-notes.tex:2515]: I think "map" would me more consistent with lecture 6 (see, e.g., page 38 [at the time of my writing this]).

**Pull request merged**